pkg/kubeutils/kubeutils.go

```diff
@@ -148,27 +147,40 @@ func WaitForInstallation(ctx context.Context, cli client.Client, writer *spinner
 	return nil
 }
 
-func writeStatusMessage(writer *spinner.MessageWriter, status embeddedclusterv1beta1.InstallationStatus) {
-	if status.State != embeddedclusterv1beta1.InstallationStatePendingChartCreation {
-		writer.Infof("Waiting for additional components to be ready: %s", status.Reason)
+func writeStatusMessage(writer *spinner.MessageWriter, install embeddedclusterv1beta1.Installation) {
+	if install.Status.State != embeddedclusterv1beta1.InstallationStatePendingChartCreation {
 		return
 	}
 
-	chartNames := ""
-	if len(status.PendingCharts) == 0 {
+	if install.Spec.Config == nil || install.Spec.Config.Extensions.Helm == nil {
 		return
-	} else if len(status.PendingCharts) == 1 {
-		// A
-		chartNames = status.PendingCharts[0]
-	} else if len(status.PendingCharts) == 2 {
-		// A and B
-		chartNames = strings.Join(status.PendingCharts, " and ")
-	} else {
-		// A, B, and C
-		chartNames = strings.Join(status.PendingCharts[:len(status.PendingCharts)-1], ", ") + " and " + status.PendingCharts[len(status.PendingCharts)-1]
+	}
+	numDesiredCharts := len(install.Spec.Config.Extensions.Helm.Charts)
+
+	// remove 'admin-console' from this
+	pendingChartsMap := map[string]struct{}{}
+	for _, chartName := range install.Status.PendingCharts {
+		pendingChartsMap[chartName] = struct{}{}
+	}
+
+	numPendingCharts := 0
+	for _, ch := range install.Spec.Config.Extensions.Helm.Charts {
+		if _, ok := pendingChartsMap[ch.Name]; ok {
+			numPendingCharts++
+		}
 	}
 
-	writer.Infof("Waiting for additional components %s to be ready", chartNames)
+	numCompletedCharts := 0
+	if numDesiredCharts-numPendingCharts > 0 {
+		// it is possible that infra charts are present here in the pending chart count, so we need to make sure this is not negative
+		numCompletedCharts = numDesiredCharts - numPendingCharts
+	}
+
+	if numCompletedCharts < numDesiredCharts {
```

Comment:
ditto

Reply:
yes, this happens when all of the charts are completed but admin console is still applying

Reply:
ah right